packages/block-editor/src/components/block-list/block-outline.native.js

```diff
@@ -44,19 +39,39 @@ function BlockOutline( {
 		hasBlockTextCategory && styles.solidBorderTextContent,
 	];
 
-	const shoudlShowOutline =
-		isSelected &&
-		( ( hasBlockTextCategory && hasInnerBlocks ) ||
-			( ! hasBlockTextCategory && hasInnerBlocks ) ||
-			( ! hasBlockTextCategory && isRootList ) ||
-			socialBlockWithOutline ||
-			textBlockWithOutline );
-
-	return (
-		shoudlShowOutline && (
-			<View pointerEvents="box-none" style={ styleSolidBorder } />
-		)
-	);
```

Comment:
While I started out addressing https://github.com/WordPress/gutenberg/issues/59380, I encountered additional block outline oddities along the way during testing. That led me to consider refactoring this code further to address an oddity — specifically, https://github.com/WordPress/gutenberg/pull/59415/commits/7fc46d684f305f6a7d2182fdb05998b22019a3bd. 

Ultimately, I began refactoring the entirety of this logic in https://github.com/WordPress/gutenberg/pull/59415/commits/ef94d34a8a6605aef0088cdbc5566c605a2c686d in an effort to reduce its complexity. Further refactors and fixes followed. The final form can be seen here in the pull request diff. 

Obviously, complexity is somewhat subjective. So, I welcome feedback on the current approach and whether we should/prefer to reinstate any of the original code structure. The automated tests should help with any future refactors. 

@geriux I'd like to request your review specifically as one of the original authors of this code. 🙇🏻